### DISCUSSION TITLE
Downstream testing

### DISCUSSION THREAD

**Discussion by trexfeathers** [SciTools/iris#5936] [Ideas]
We recently merged #5801, which had a full set of passing CI. However the feature then broke Iris-grib, without an obvious simple fix. Because the PR is already merged, this is no longer the author's problem but is instead the responsibility of Iris-grib maintainers.

We have an ambition to have more parts of Iris separated and optional - #4798 - however what the above demonstrates is that this could potentially shift work from contributors onto maintainers. Maintainer workload is agreed to be one of the biggest threats to open source sustainability, so I see this prospect as a significant problem.

I am therefore keen that we think about ways of running tests for Iris satellite projects (e.g. Iris-grib, iris-esmf-regrid) WITHIN Iris pull requests, as a way of alerting authors and reviewers that their work may cause downstream problems and give them an opportunity to address those.

**Comment by pp-mo** [comment]
There is now an interesting example [here in Iris](https://github.com/SciTools/iris/blob/98193c013bd1812cb91610c24c27bce0ad5e72c2/lib/iris/tests/unit/fileformats/netcdf/attribute_handlers/test_GribParamHandler.py#L9), and [here in iris-grib](https://github.com/SciTools/iris-grib/blob/cf1b5e4e710bd663c8650aaaf5cabebc76ce2ba1/src/iris_grib/tests/integration/iris_integration/test_nc_attribute_handlers.py), as in   https://github.com/SciTools/iris-grib/pull/677
 - these tests are written in Iris, but only run on iris-grib CI (to avoid installing iris-grib when testing iris)
